### PR TITLE
feat: add electron app for running on desktop

### LINKED ISSUE #31: Electron integration
Let's have an Electron app to run the UI.

We did something similar in https://github.com/nearform/brokeneck/, check out how it's done there.

### PR DESCRIPTION
**Pull request by gilach**
Closes #31

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,14 @@
 .vscode/
+.idea/
 .nyc_output/
 .test_results/
 node_modules/
 coverage/
 
+**/package-lock.json
```
Comment (simoneb) [inline .gitignore:8]: why this exclusion?

File: .gitignore [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
+
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+**/dist/
```
Comment (simoneb) [inline .gitignore:14]: stuff that's specific to a package, let's put it in the package's gitignore so the root doesn't need to be aware of specifics of a nested package

File: packages/autocannon-ui-desktop/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+  "extends": ["standard", "plugin:prettier/recommended"],
```
Comment (simoneb) [inline packages/autocannon-ui-desktop/.eslintrc:2]: just checking if we're using standard elsewhere. if not, let's remove it

File: packages/autocannon-ui-desktop/main.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  const address = await startServer()
+  await app.whenReady()
+  await createWindow(address)
+}
+
+async function startServer() {
+  const fastify = Fastify()
+  fastify.register(autocannonUiBackend)
+
+  require.resolve('autocannon-ui-frontend')
```
Comment (simoneb) [inline packages/autocannon-ui-desktop/main.js:24]: remove

File: packages/autocannon-ui-frontend/package.json [hunk 5]
```diff
@@ -10,6 +10,9 @@
     "_test:qa": "mocha --timeout=30000 test/autocannon.test.js",
     "test:qa": "wait-on tcp:3000 && yarn _test:qa"
   },
+  "files": [
+    "dist"
+  ],
```
Comment (simoneb) [inline packages/autocannon-ui-frontend/package.json:15]: what is this needed for?

File: packages/autocannon-ui-desktop/main.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      contextIsolation: true,
+      preload: path.join(app.getAppPath(), 'preload.js'),
+      additionalArguments: [JSON.stringify(store.store)]
+    }
+  })
+
+  await win.loadURL(address)
+
+  win.on('close', async () => {
+    const localStorage = await win.webContents.executeJavaScript(
```
Comment (simoneb) [inline packages/autocannon-ui-desktop/main.js:55]: is this needed? what for?

**Review by simoneb — COMMENTED** [review]
let's put a comment in the empty index.js file of the frontend to explain why it's there

File: packages/autocannon-ui-desktop/main.js (hunk 6 above)
Comment (gilach) [reply packages/autocannon-ui-desktop/main.js:55]: currently not, but in the future it will be helpful for saving config in local storage

Comment (simoneb) [reply packages/autocannon-ui-desktop/main.js:55]: okay fair enough

**Review by simoneb — APPROVED** [review]

**Pull request merged**